src/components/application_manager/src/event_engine/event_dispatcher_impl.cc

```diff
@@ -118,7 +115,10 @@ void EventDispatcherImpl::remove_observer(const Event::EventID& event_id,
 }
 
 void EventDispatcherImpl::remove_observer(EventObserver& observer) {
-  remove_observer_from_vector(observer);
+  {
+    AutoLock auto_lock(observer_lock_);
+    remove_observer_from_vector(observer);
```

Comment:
@OlhaVorobiova this should not be scoped as you are accessing `observers_event_` on the next line